mrq/job.py

```diff
@@ -274,7 +279,22 @@ def perform(self):
 
         self.task.is_main_task = True
 
-        result = self.task.run_wrapped(self.data["params"])
+        try:
+            if self.task.max_concurrency:
+                pipeline = context.connections.redis.pipeline()
+                pipeline.incr(self.redis_concurrency_key)
+                pipeline.expireat(self.redis_concurrency_key, int(time.time()) + self.timeout)
+                current = pipeline.execute()[0]
```

Comment:
this is not safe enough - with maxconcurrency=2, if a worker dies here, the count will stay at a minimum at 1, and if there are still tasks happening, the key will never expire.